src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala

```diff
@@ -30,6 +30,38 @@ trait Enclosing { this: TypeInfoImpl =>
     case _ => id
   })
 
+  lazy val enclosingLoop: PNode => Option[PForStmt] = {
+    down[Option[PForStmt]](None){ case x: PForStmt => Some(x) }
+  }
+
+  def enclosingLabeledLoop(label: PLabelUse, node: PNode) : (Option[PForStmt], Vector[PExpression]) =
+    enclosingLoop(node) match {
+      case None => (None, Vector.empty)
+      case Some(l) => {
+        val p = tree.parent(l).head
+        if (p.isInstanceOf[PLabeledStmt]) {
+          val loopLabel = p.asInstanceOf[PLabeledStmt].label.name
```

Comment:
I would avoid doing `isInstanceOf[...]` followed by `asInstanceOf[...]` and just do pattern matching when that is practical